### ISSUE TITLE
[question] which (mobile) browsers support OpenSearch?

### ISSUE THREAD

**Issue by return42** [searxng/searxng#431]
[Autodiscovery of search plugins](https://developer.mozilla.org/en-US/docs/Web/OpenSearch#autodiscovery_of_search_plugins) gives the user the ability to easily add alternative (and privacy-respecting) search engines, including autocompletion (and other) features.

Common desktop browsers (e.g. [Firefox](https://support.mozilla.org/en-US/kb/add-or-remove-search-engine-firefox),  [Chromium](https://www.chromium.org/tab-to-search) or [MS-Edge](https://support.microsoft.com/en-us/microsoft-edge/change-your-default-search-engine-f863c519-5994-a8ed-6859-00fbc123b782)) offer [Autodiscovery of search plugins](https://developer.mozilla.org/en-US/docs/Web/OpenSearch#autodiscovery_of_search_plugins) based on the [OpenSearch protocol](https://github.com/dewitt/opensearch).  But it is very hard to find a mobile browser that can do that too ...

I opened this issue to collect more informations about browser that support OpenSearch, especially I am looking for moobile (Android) alternatives, since my favourite browser sadly does not [[1]](https://github.com/mozilla-mobile/fenix/issues/21883) [[2]](https://github.com/mozilla-mobile/fenix/issues/8889) [[3]](https://f-droid.org/en/packages/org.mozilla.fennec_fdroid/) [[4]](https://github.com/mozilla-mobile/fenix/issues/6615#issuecomment-1027947583).

The intention of this issue is to collect some recommendations we can add to our documentation / hints are welcome ...

Related:

- https://github.com/searxng/searxng/pull/1655
- https://github.com/searxng/searxng/discussions/1288
- https://github.com/searxng/searxng/issues/1613#issuecomment-1214293533
- POST vs GET --> some broswers don't like HTTP POST requests in a OpenSearch query / e.g. Chrome on iOS : https://app.element.io/#/room/#searxng:matrix.org/$4_hkVaUiVBW93F2uh8YcDY3J3SKIh5ERs8pkUqyQP0E
- https://github.com/searxng/searxng/issues/2086#issuecomment-1374804339
- https://github.com/searxng/searxng/discussions/1866
- https://github.com/searxng/searxng/discussions/2133#discussioncomment-4956291
- https://github.com/searxng/searxng/discussions/1288
- https://github.com/searxng/searxng/issues/2190
- https://github.com/searxng/searxng/issues/546
- [Vavalid does not offer Autodiscovery of search plugins based on the OpenSearch protocol](https://github.com/searxng/searxng/issues/2688#issuecomment-1695768989)

**Comment by dalf** [comment]
It should be close / identical to what whoogle supports : https://github.com/benbusby/whoogle-search#set-whoogle-as-your-primary-search-engine

**Comment by d-g** [comment]
Just as I have noted at [1], I am rather surprised, that you are unaware that, that Chromium-based browsers, i. e. _overwhelming majority_ of browser instances used at Android, support it out of a box, so Iʼll reiterate how to use it here as well:

1. You navigate to, say, https://searx.be/ (to the _root_ page).
2. Perform any search.
3. Go to Settings / Search Engine / Recently Visited and now you can make it a default search.

Alas, there is no any other search but default (i. e. no icons to choose an alternative one, no keywords), but that was not your issue.

[1] https://github.com/mozilla-mobile/fenix/issues/21883#issuecomment-962210700

**Comment by return42** [comment]
> I am rather surprised, that you are unaware that, that Chromium-based browsers, i. e. _overwhelming majority_ of browser instances used at Android, support it out of a box

I'm aware of Chromium-based browsers on android devices, but they are full of bloatware and tracking features.  Sorry if it was not clear; this project is about _"not to be tracked"_.  In this issue I ask about alternatives without tracking & bloatware crap. E.g. Apps available on free Apps-stores like FDroid / I do not have a google account and I do not want to be a part of this mindless and ruthless market.

**Comment by return42** [comment]
I haven't had time to test, but [bromite](https://github.com/bromite/bromite/tree/master) might be an alternative based on Chromium.

**Comment by vityav** [comment]
Coming from the linked Firefox issue, I installed and briefly tested Firefox, Chrome, Vivaldi, Opera, Brave, Edge, Kiwi, DuckDuckGo, and Bromite. The only ones with the ability to add custom search engines from OpenSearch are the ones closest to stock Chromium (Chrome, Kiwi, and Bromite). I should note that in Bromite you also have to go into privacy settings to enable autocomplete/suggest.

**Comment by return42** [comment]
@vityav wow, thanks for testing such a wide range of browsers! .. I haven't had time for right now .. thanks for sharing your experience here.

**Comment by catfluoride** [comment]
> Coming from the linked Firefox issue, I installed and briefly tested Firefox, Chrome, Vivaldi, Opera, Brave, Edge, Kiwi, DuckDuckGo, and Bromite. The only ones with the ability to add custom search engines from OpenSearch are the ones closest to stock Chromium (Chrome, Kiwi, and Bromite). I should note that in Bromite you also have to go into privacy settings to enable autocomplete/suggest.


Hi! I was able to use SearX on Brave for Android until at least version 1.25. After that, no way to make searx appear among the "Recently visited" engines. Desktop version works nicely, instead.

Interestingly, **if I set SearX as the default SE on Brave 1.25** and then update it, SearX keeps appearing among the "Recently visited" search engines and works correctly.

I also opened an issue on Brave's GitHub: https://github.com/brave/brave-browser/issues/19679

**Comment by return42** [comment]
Just want to leave a link to https://github.com/searxng/searxng/issues/546#issuecomment-983818436

**Comment by return42** [comment]
In [bromite](https://github.com/bromite/bromite#features-not-related-to-privacy)'s feature list I read "allow OpenSearch search engine detection with paths"

Does anyone have any experience with Bromite on Android; what are your use-cases and setups?

@PetbkA and @catfluoride in https://github.com/searxng/searxng/issues/1613#issuecomment-1214293533 I see you are using Bromite .. would you like to share your experience and motivations (here in this thread) / thanks!

**Comment by PetbkA** [comment]
@return42 , I'm using Bromite for testing purposes only due to inconvenient UI, inability to add custom search engines by url and change engines in address bar. My main browser is [Berry](https://play.google.com/store/apps/details?id=jp.ejimax.berrybrowser&gl=US).

I did quick testing with:
1. Bromite 104.0.5112.63;
2. Chrome 104.0.5112.69
on YouTube (Firefox discovers its search engine so I guess it's using OpenSearch) - no new engines appeared

**Comment by return42** [comment]
> I did quick testing with:

@PetbkA I am hard to grasp today .. about your comment; does it mean SearXNG's OpenSearch does work in these browsers or do you have issues?

**Comment by PetbkA** [comment]
> > I did quick testing with:
> 
> @PetbkA I am hard to grasp today .. about your comment; does it mean SearXNG's OpenSearch does work in these browsers or do you have issues?

I thought the discussion here is only about _implementing_ OpenSearch support so didn't tested SearxNG instances.

Tested PaulGO and serx.ml instances with Bromite and it seems that they're really being
[added](https://user-images.githubusercontent.com/22843816/184531737-a0f804ff-46b0-47fa-a704-94090648fbf4.png) as search engines, neverthless of _Autocomplete searches and URLs_ option state

**Comment by return42** [comment]
> I thought the discussion here is only about _implementing_ OpenSearch support so didn't tested SearxNG instances.

Thats true .. but the background of this thread is more or less if a browser supports 

- [OpenSearch](https://github.com/dewitt/opensearch/blob/master/opensearch-1-1-draft-6.md) (the opensearch.xml file) and
- [Autodiscovery of search plugins](https://developer.mozilla.org/en-US/docs/Web/OpenSearch#autodiscovery_of_search_plugins) 

Both components have to be implemented in the browser .. and this is where Browsers differ; some do not have any support, other implemented both but have its own quirks.  

That (how a browser implements Autodiscovery & the opensearch.xml file) is why you can add e.g yoahoo in browser "A" but not in browser "B" .. while you can add e.g. qwant to "A" and "B" (same with adding SearXNG).

We at SearXNG try to find solutions that fits to as many browsers as possible.

**Comment by catfluoride** [comment]
> In [bromite](https://github.com/bromite/bromite#features-not-related-to-privacy)'s feature list I read "allow OpenSearch search engine detection with paths"
> 
> Does anyone have any experience with Bromite on Android; what are your use-cases and setups?
> 
> @PetbkA and @catfluoride in [#1613 (comment)](https://github.com/searxng/searxng/issues/1613#issuecomment-1214293533) I see you are using Bromite .. would you like to share your experience and motivations (here in this thread) / thanks!

Hi! My comment was about the "search" action not being triggered by the return button on my android keyboard, using Brave browser. I don't know what has changed in the meanwhile, but now it works correctly.
